pyqtorch/core/utils.py

```diff
@@ -86,23 +85,20 @@ def _apply_gate(
     - state (torch.Tensor): the quantum state after application of the gate.
     Same shape as `ìnput_state`
     """
-    mat = mat.reshape([2] * len(qubits) * 2)
+    mat = mat.view([2] * len(qubits) * 2)
```

Comment:
why view over reshape?

Reply:
view just ensures that the tensor is not copied. but i think that .reshape does view under the hood if it can in very recent versions of torch

Reply:
I guess for gates we're quite confident that they will always be viewable and contigous, but I'm not sure that it's necessarily the case for Hamiltonians. Are they also using apply_gate? If so, do we think that they'll always be viewable?

This SO post agrees with what I remember:
https://stackoverflow.com/questions/49643225/whats-the-difference-between-reshape-and-view-in-pytorch

Reply:
they use _apply_batch_gate which uses `einsum` and that reshapes under the hood, not sure via allocating copies though, if it can avoid it, then yes but i guess with varying qubit support it has to rearrange the tensor which makes it not contiguous anymore, thats btw a big argument for not using einsum